parser/src/main/java/me/vertretungsplan/parser/LegionBoardParser.java

```diff
@@ -0,0 +1,347 @@
+			final HashSet<String> classes = new HashSet<>();
+			classes.add(singleClass);
+			substitution.setClasses(classes);
+		}
+		// Set type
+		String type = "Unknown";
+		switch (change.getString("type")) {
+			case "0":
+				type = "Entfall";
+				break;
+			case "1":
+				type = "Vertretung";
+				break;
+			case "2":
+				type = "Information";
+				break;
+		}
+		substitution.setType(type);
+		// Set color
+		substitution.setColor(colorProvider.getColor(type));
+		// Set covering teacher
+		final String coveringTeacherId = change.getString("coveringTeacher");
+		if (!coveringTeacherId.equals("0")) {
+			if (teachersHashMap == null) {
+				throw new IOException("Change references a covering teacher but teachers are empty.");
+			}
+			substitution.setTeacher(teachersHashMap.get(coveringTeacherId));
+		}
+		// Set teacher
+		final String teacherId = change.getString("teacher");
+		if (!teacherId.equals("0")) {
+			if (teachersHashMap == null) {
+				throw new IOException("Change references a teacher but teachers are empty.");
+			}
+			if (type.equals("Vertretung") || !coveringTeacherId.equals("0")) {
+				substitution.setPreviousTeacher(teachersHashMap.get(teacherId));
+			}
+			else {
+				substitution.setTeacher(teachersHashMap.get(teacherId));
+			}
+				
+		}
+		// Set description
+		substitution.setDesc(change.getString("text"));
+		// Set lesson
+		final String startingHour = change.getString("startingHour");
+		final String endingHour = change.getString("endingHour");
+		if (!startingHour.equals("") || !endingHour.equals("")) {
+			String lesson = "";
+			if (!startingHour.equals("") && endingHour.equals("")) {
+				lesson = "Ab " + startingHour;
+			}
+			if (startingHour.equals("") && !endingHour.equals("")) {
+				lesson = "Bis " + endingHour;
+			}
+			if (!startingHour.equals("") && !endingHour.equals("")) {
+				lesson = startingHour + " - " + endingHour;
+			}
+			substitution.setLesson(lesson);
+		}
+		return substitution;
+	}
+
+	/**
+	 * Sorts a JSONArray by date.
+	 * Source: http://stackoverflow.com/a/19546513
+	 */
+	private JSONArray sortByDate(JSONArray jsonArray) throws JSONException {
+		final JSONArray sortedJsonArray = new JSONArray();
+
+		final List<JSONObject> jsonValues = new ArrayList<JSONObject>();
+		for (int i = 0; i < jsonArray.length(); i++) {
+			jsonValues.add(jsonArray.getJSONObject(i));
+		}
+		Collections.sort(jsonValues, new Comparator<JSONObject>() {
+			@Override
+			public int compare(JSONObject firstJsonObject, JSONObject secondJsonObject) {
+				try {
+					final SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd");
+					final Date firstDate = simpleDateFormat.parse(firstJsonObject.getString("startingDate"));
+					final Date secondDate = simpleDateFormat.parse(secondJsonObject.getString("startingDate"));
+					return firstDate.compareTo(secondDate);
+				} catch (ParseException e) {
+					e.printStackTrace();
+				} catch (JSONException e) {
+					e.printStackTrace();
+				}
```

Comment:
you can collapse the two `catch` branches to
```
} catch (ParseException | JSONException e) {
```
as long as you do the same thing in both.